File: Backend/src/main/java/com/hotelLosViejos/HotelLosViejos/Dominio/Reserva.java
package com.hotelLosViejos.HotelLosViejos.Dominio;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class  Reserva {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String numeroReserva;

    private LocalDateTime fechaLlegada;

    private LocalDateTime fechaSalida;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "habitacion_id")
    private Habitacion habitacion;
}

File: Frontend/types/Reserva.ts
import type { ClienteBase } from './Cliente';
import type { HabitacionBase } from './Habitacion';
export interface ReservaPayload {
    nombre: string
    apellidos: string
    correo: string
    numeroTarjeta: string
    habitacionId: number
    fechaLlegada: string 
    fechaSalida: string
  }

export interface ReservaLectura {
  id: number,
  numeroReserva: string,
  fechaLlegada: string,
  fechaSalida: string,
  cliente: ClienteBase,
  habitacion: HabitacionBase
}